src/elements.rs

```diff
@@ -451,6 +451,11 @@ pub struct TrackEntry {
 pub fn track_entry(input: &[u8]) -> IResult<&[u8], TrackEntry, Error> {
     ebml_master(0xAE, |i| {
         map(
+            // Because the Permutation trait is implemented for tuples of up
+            // to 21 elements, it is necessary to launch the permutation
+            // function twice. The second one acts on the remaining input
+            // produced by the first function.
+            // https://docs.rs/nom/latest/nom/branch/trait.Permutation.html
```

Comment:
Does that actually do the right thing then? It would first consume all the things from the first part of the pair, then all the things from the second part. If an ID of the first comes after an ID of the second then it would simply be skipped or considered an error, or not?

Might be better to extend `permutation` to work with bigger tuples. Too bad we don't have variadic tuples in Rust yet...

Reply:
Yes, we can occur in that problem. The case you have described would lead to an error. We have to expand `permutation` to work with bigger tuples. Can't we occur in performance problems with 41 tuples?

Reply:
`permutation` is implemented as a loop with a match https://github.com/Geal/nom/blob/main/src/branch/mod.rs#L224 that analyzes the result of each parser, so I think we can replicate the same code

Reply:
In that case just implement it in nom for more than 21-tuples :)